src/parcsr_ls/par_cycle.c

```diff
@@ -286,6 +286,15 @@ hypre_BoomerAMGCycle( void              *amg_vdata,
    hypre_GpuProfilingPushRange(nvtx_name);
    while (Not_Finished)
    {
+#ifdef HYPRE_USING_NODE_AWARE_MPI
+      if (level >= hypre_HandleNodeAwareSwitchoverThreshold(hypre_handle()))
+      {
+         hypre_HandleUsingNodeAwareMPI(hypre_handle()) = 1;
```

Comment:
@liruipeng however, this other global variable cannot. One idea is to move it to the comm_pkg struct.

Reply:
I think to do that, comm_pkg would need to be accessible within `hypre_BoomerAMGCycle` which I don't think it is unless I'm mistaken. I don't see it in hypre_ParAMGData. Maybe it could be added as a field there but would need to be properly set and maintained.

Reply:
You can use https://github.com/hypre-space/hypre/blob/36ab29ba0977294a741987c4f921fe02840dff59/src/parcsr_mv/par_csr_matrix.h#L108

Reply:
I could be wrong...but to what granularity do we want to control `hypre_HandleUsingNodeAwareMPI`? If we put in commPkg, it would be per commPkg/Parcsrmatrix. I thought it's a more global thing, like the whole AMG solver. Again, putting it into the AMG struct is ideal, if we can find a clean way to do it.

Reply:
As far as I can see, the node aware option has been implemented only for the IJ interface. If we want it on the other interfaces in the future, we could extend their respective comm_pkg structs to handle this option.

Note I'm suggesting to put `using_node_aware_mpi` under the comm_pkg struct here, i.e., not under the AMG struct as for the other case :)

Reply:
This may still work but it's my understanding that we have multiple matrices per level that each have a comm pkg. The `using_node_aware` toggle logically applies to a whole level with multiple matrices. So I think we would need to get the comm pkg for A and P and set them both to use node aware in their comm pkgs but I think that would work. What do y'all think?

Reply:
Thanks @geraldc-unm! I spoke to Rui Peng and shared that's the right approach in my opinion. Let's think about this (moving the variable to inside comm_pkg) a bit more...